### PR TITLE
parser/mysql, privilege/privileges: fix trigger privilege

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32891

Problem Summary:

A grant statement was failing because of the TRIGGER privilege, which in MySQL can be specified on a DB and Table level. This addresses the reported issue, but a second issue where a nil pointer dereference was found did not seem to be reproducible against master. It might have just been my feature branch that failed, but I've added the case to the tests in case my feature branch merges and we miss it.

### What is changed and how it works?

The trigger privilege is now specifyable on a DB or Table level. This has no effect in TiDB as triggers are not supported, but it improves MySQL compatibility.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

```release-note
The trigger privilege is now specifyable on a DB or Table level. This has no effect in TiDB as triggers are not supported, but it improves MySQL compatibility.
```

### CODE REVIEW

File: privilege/privileges/privileges_test.go [hunk 1]
```diff
@@ -2875,6 +2875,23 @@ func TestSkipGrantTable(t *testing.T) {
...
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec(`CREATE USER ucorrect1, ucorrect2;`)
+	tk.MustExec(`CREATE TABLE test.trigger_table (a int)`)
+	tk.MustExec(`GRANT CREATE TEMPORARY TABLES,DELETE,EXECUTE,INSERT,SELECT,SHOW VIEW,TRIGGER,UPDATE ON test.* TO ucorrect1;`)
+	tk.MustExec(`GRANT TRIGGER ON test.trigger_table TO ucorrect2;`)
+	tk.MustExec(`DROP TABLE test.trigger_table`)
+
+	err := tk.ExecToErr(`GRANT CREATE TEMPORARY TABLES,DELETE,EXECUTE,INSERT,SELECT,SHOW VIEW,TRIGGER,UPDATE ON test.* TO uincorrect;`)
+	require.EqualError(t, err, "[executor:1410]You are not allowed to create a user with GRANT")
+
```
Comment (bb7133) [inline privilege/privileges/privileges_test.go:2892]: Please remove this blank line.

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**